case.h

```diff
@@ -1,9 +1,12 @@
 #ifndef CASE_H
 #define CASE_H
 
+#include <string.h>
+#include <strings.h>
+
 extern void case_lowers();
 extern void case_lowerb();
-extern int case_diffs();
+#define case_diffs(s,t) (strcasecmp((s),(t)) != 0)
```

Comment:
Why `strncmp()` elsewhere but not `strncasecmp()` here, or vice versa? Serious question. When do we trust non-`n` functions, when would we prefer to call `strlen()` and pass that in, and when would we prefer to already have the length and pass _that_ in?

Reply:
because ```strncasecmp()``` is nonstandard and may not be available everywhere.

Reply:
Is `strcasecmp` not historically one of the sometimes dangerous libc string functions? I don’t know which those were and I don’t know how to find out, either.

Reply:
Those like ```strcpy()``` where there is possibility of overrun or ```strncpy()``` where the terminating 0-byte may not be written.